### PR TITLE
Make uv and nng private and static (#323)

### PR DESCRIPTION
**Pull request by broskoTT**
### Issue
Original PR was #323, then the reverted one was #336

### Description
libnng.so is no longer a public dependency as of this PR: https://github.com/tenstorrent/tt-umd/pull/224
Additionally, given that it is only consumed here it would be advantageous to statically link it.
Why?
So we don't have to:
- Install it
- Package it in a debian
- Bundle it in a ttnn python wheel ... etc ... 

Along the above lines, I don't want to run into issues like this:
```
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 185, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.8/runpy.py", line [11](https://github.com/tenstorrent/tt-metal/actions/runs/11964313949/job/33356936955#step:6:12)1, in _get_module_details
    __import__(pkg_name)
  File "/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/tests/end_to_end_tests/env/lib/python3.8/site-packages/ttnn/__init__.py", line 20, in <module>
    import ttnn._ttnn
ImportError: libnng.so.1: cannot open shared object file: No such file or directory
```
### List of the changes
Build nng as static
Make it private linkage

### Testing
Hoping CI is good enough

### API Changes
There are no API changes in this PR.

### CODE REVIEW

File: device/CMakeLists.txt [hunk 1]
```diff
@@ -41,16 +41,15 @@ target_include_directories(
 )
 
 # flatbuffers is public - exposed to tt_metal by tt_simulation_device_generated.h
-# nng is public - exposed to tt_metal by tt_simulation_host.hpp
 target_link_libraries(
     device
     PUBLIC
-        nng
```
Comment (broskoTT) [inline device/CMakeLists.txt:48]: @blozano-tt Please make sure that tt_metal CI passes with this change. Copying the comment from the revert PR:

This change broke tt-metal build.
post commit workflow failing: https://github.com/tenstorrent/tt-metal/actions/runs/12031721201
post commit workflow on this branch: https://github.com/tenstorrent/tt-metal/actions/runs/12032379476/job/33545486354
blackhole post-commit failing: https://github.com/tenstorrent/tt-metal/actions/runs/12031811999
blackhole post-commit on this branch: https://github.com/tenstorrent/tt-metal/actions/runs/12032509241

**Review by blozano-tt — APPROVED** [review]
LGTM!

File: device/CMakeLists.txt (hunk 1 above)
Comment (blozano-tt) [reply device/CMakeLists.txt:48]: Issues seem to be fixed here:


https://github.com/tenstorrent/tt-metal/actions/runs/12113533333

Comment (blozano-tt) [reply device/CMakeLists.txt:48]: https://github.com/tenstorrent/tt-metal/issues/15527#issuecomment-2510682445

Comment (broskoTT) [reply device/CMakeLists.txt:48]: Thanks for fixing this!

**Pull request merged**